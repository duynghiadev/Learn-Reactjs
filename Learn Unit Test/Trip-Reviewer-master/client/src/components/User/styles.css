.user-container {
  max-width: 500px;
  margin: 75px auto;
}
.user-container .data-wrapper {
  display: flex;
  flex-direction: row;
  margin-bottom: 75px;
}
.user-container .avatar {
  width: 200px;
  height: 200px;
}
.user-container .info-wrapper {
  margin-left: 100px;
}
.user-container .info-item {
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 18px;
  font-weight: 300;
  margin-bottom: 20px;
  color: #a4a4a4;
}
.user-container .info-title {
  display: block;
  font-weight: 500;
  font-size: 24px;
  color: #02bcd4;
  margin-bottom: 5px;
}
.user-container .button-link {
  width: 172px;
  display: inline-block;
  font-size: 20px;
  padding: 10px 15px;
  background: #02bcd4;
  color: #fff;
  font-weight: 300;
  border-radius: 3px;
  border: none;
}
.user-container .button-link:hover {
  transform: scale(1.05);
  background: #009eb3;
}
.user-container .button-link:active {
  transform: scale(1.01);
  background: #0495a8;
}

@media (max-width: 800px) {
  .user-container {
    margin: 50px auto 75px;
  }
  .user-container .data-wrapper {
    flex-direction: column;
    align-items: center;
    text-align: center;
    margin: 0;
  }
  .user-container .info-wrapper {
    margin: 50px 0;
  }
  .user-container .data .info-item {
    text-align: center;
  }
}
