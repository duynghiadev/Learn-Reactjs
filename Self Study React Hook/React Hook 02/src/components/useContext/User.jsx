import { useContext } from 'react'
import { AppContext } from './context'

const User = () => {
  const { username } = useContext(AppContext)

  return (
    <section>
      <h1>User: {username}</h1>
    </section>
  )
}

export default User
