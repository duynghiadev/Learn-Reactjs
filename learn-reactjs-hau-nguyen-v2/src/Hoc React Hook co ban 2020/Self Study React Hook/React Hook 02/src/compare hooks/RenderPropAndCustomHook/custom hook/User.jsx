import { useState } from 'react'
import { UserContext } from './UserProfile'

function User({ children }) {
  const [user, setUser] = useState({
    name: 'Aditya',
    email: 'adityaa803@gmail.com'
  })

  const value = { user, setUser }

  return (
    <>
      <UserContext.Provider value={value}>{children}</UserContext.Provider>
    </>
  )
}

export default User
