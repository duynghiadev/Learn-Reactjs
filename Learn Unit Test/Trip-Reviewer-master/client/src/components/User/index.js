import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import UserAvatar from '../UserAvatar';
import { EDIT_PROFILE } from '../../constants/routes';

import './styles.css';

const User = props => {
  const user = props.users.login;
  const redirect = `${EDIT_PROFILE}`;

  return (
    <div className="user-container">
      <div className="data-wrapper">
        <div className="avatar">
          <UserAvatar filename={user.avatar} />
        </div>

        <div className="info-wrapper">
          <div className="info-item" title={user.name}>
            <span className="info-title">Name:</span>
            {user.name}
          </div>
          <div className="info-item" title={user.lastname}>
            <span className="info-title">Lastname:</span>
            {user.lastname}
          </div>
          <div className="info-item" title={user.email}>
            <span className="info-title">Email:</span>
            {user.email}
          </div>
        </div>
      </div>

      <div className="text-center button-wrapper">
        <Link to={redirect} className="button-link">
          Edit Profile
        </Link>
      </div>
    </div>
  );
};

User.propTypes = {
  users: PropTypes.object.isRequired,
};

export default User;
