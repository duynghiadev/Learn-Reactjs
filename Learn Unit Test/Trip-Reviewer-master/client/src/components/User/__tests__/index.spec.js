import React from 'react';
import { create } from 'react-test-renderer';

import User from '..';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));
jest.mock('../../UserAvatar', () => 'UserAvatar');

const mockComponent = props => <User {...props} />;

describe('<User />', () => {
  it('should render component', () => {
    const props = {
      users: {
        login: {
          name: 'name',
          lastname: 'lastname',
          email: 'email',
        },
      },
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
