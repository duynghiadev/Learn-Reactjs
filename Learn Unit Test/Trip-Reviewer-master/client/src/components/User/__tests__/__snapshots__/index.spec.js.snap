// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> should render component 1`] = `
<div
  className="user-container"
>
  <div
    className="data-wrapper"
  >
    <div
      className="avatar"
    >
      <UserAvatar
        filename={undefined}
      />
    </div>
    <div
      className="info-wrapper"
    >
      <div
        className="info-item"
        title="name"
      >
        <span
          className="info-title"
        >
          Name:
        </span>
        name
      </div>
      <div
        className="info-item"
        title="lastname"
      >
        <span
          className="info-title"
        >
          Lastname:
        </span>
        lastname
      </div>
      <div
        className="info-item"
        title="email"
      >
        <span
          className="info-title"
        >
          Email:
        </span>
        email
      </div>
    </div>
  </div>
  <div
    className="text-center button-wrapper"
  >
    <Link
      className="button-link"
      to="/user/edit-profile"
    >
      Edit Profile
    </Link>
  </div>
</div>
`;
